### PR TITLE
refactor(FOROME-1006): dtree controls

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1006

In addition:
1. Added validation for creation filter (dtree/preset) name
2. Fixed truncation for selected filter (dtree/preset) name

### CODE REVIEW

**Comment by AndreyUstyumenko** [comment]
Would like to learn what exactly has been done here.
The task has no description, although it has [Redesign] prefix, so I assume there is supposed to be job related to components, however I see refactoring traces as well. Please add maybe at least screenshots in the task description to understand what's been redesigned.

File: src/pages/filter/dtree/components/filter-control-dtree/filter-control-dtree.utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import filterDtreesStore from '@store/filter-dtrees'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/dtree/components/filter-control-dtree/filter-control-dtree.utils.ts]: Why create functions that do nothing but calling another functions? Call store methods directly instead.

File: src/service-providers/decision-trees/decision-trees.interface.ts [hunk 2]
```diff
@@ -197,3 +197,15 @@ export type TGetFullDtreeStatParams = {
...
+  ds: string
+  code: string
+  instr: TDtreeModifyingActions
+}
+
+export type TDeleteDecisionTreeParams = {
+  ds: string
+  code: string
+  instr: TDtreeModifyingActions
+}
```
Comment (AndreyUstyumenko) [inline src/service-providers/decision-trees/decision-trees.interface.ts:211]: looks like interfaces to me

File: src/store/dtree.ts [hunk 3]
```diff
@@ -309,6 +327,55 @@ class DtreeStore {
...
+      })
+      .finally(() => {
+        runInAction(() => {
+          this.isDtreeLoading = false
+        })
+      })
+  }
+
+  private loadDtree(dtreeName: string): void {
+    this.updateDtreeCode(
```
Comment (AndreyUstyumenko) [inline src/store/dtree.ts]: These two functions look strange. Why break `then` and `catch` onto different methods? Is `updateDtreeCode` supposed to be reusable? But it's private.

File: src/store/filter-dtrees/filter-dtrees.store.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  public get isFetchingDtrees(): boolean {
+    return this.dtrees.isFetching
+  }
+
+  public get availableDtrees(): ISolutionEntryDescription[] {
+    return toJS(this.dtrees.data) ?? []
+  }
+
+  public get activeDtree(): string {
+    return this._activeDtree
```
Comment (AndreyUstyumenko) [inline src/store/filter-dtrees/filter-dtrees.store.ts]: just make it public

File: src/store/filter-dtrees/filter-dtrees.store.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      .then(
+        () => {
+          this.dtrees.invalidate()
+          this.setActiveDtree(dtreeName)
+          showToast(
+            t('header.dtreeAction.success.create', { dtreeName }),
+            'success',
+          )
+        },
+        () => {
```
Comment (AndreyUstyumenko) [inline src/store/filter-dtrees/filter-dtrees.store.ts]: move it to `catch`

File: src/store/dtree.ts (hunk 3 above)
Comment (Lapk1n) [reply src/store/dtree.ts]: I thought of `joinDtree` probably, but there is no such a function for dtree

File: src/components/preset-control/preset-control-button/preset-control-button.module.css [hunk 6]
```diff
@@ -12,6 +12,7 @@
 
   &__label {
     flex: 1 0 auto;
+    width: 85%;
```
Comment (AlMaQntr) [inline src/components/preset-control/preset-control-button/preset-control-button.module.css]: Why you use percentage value? May be right margin more would be more appropriate?

File: src/components/preset-control/preset-create-dialog/preset-create-dialog.tsx [hunk 7]
```diff
@@ -1,37 +1,65 @@
...
   isOpen,
+  controlName,
   onClose,
   onCreate,
 }: IPresetCreateDialogProps): ReactElement => {
-  const [presetName, setPresetName] = useState('')
+  const [filterName, setFilterName] = useState('')
+  const [error, setError] = useState<boolean>(false)
+
+  useEffect(() => {
```
Comment (AlMaQntr) [inline src/components/preset-control/preset-create-dialog/preset-create-dialog.tsx]: Don't use useState+useEffect for this. Use useMemo
```
const error = useMemo(() => filters.some(...), [...])
```

File: src/core/enum/filter-control-enum.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export enum FilterControl {
```
Comment (AlMaQntr) [inline src/core/enum/filter-control-enum.ts:1]: I think is better to put this strings in i18n

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by AlMaQntr — APPROVED** [review]

File: src/components/preset-control/preset-control-button/preset-control-button.module.css (hunk 6 above)
Comment (Lapk1n) [reply src/components/preset-control/preset-control-button/preset-control-button.module.css]: Screenshot 2022-05-18 at 11 19 24
For truncation purposes

File: src/components/preset-control/preset-create-dialog/preset-create-dialog.tsx (hunk 7 above)
Comment (Lapk1n) [reply src/components/preset-control/preset-create-dialog/preset-create-dialog.tsx]: cool

File: src/components/preset-control/preset-control-button/preset-control-button.module.css (hunk 6 above)
Comment (AlMaQntr) [reply src/components/preset-control/preset-control-button/preset-control-button.module.css]: Yes, but why in percents)? May be right margin with fixed value?

**Review by QYaroslavTrefilov — APPROVED** [review]

**Pull request merged**